### ISSUE TITLE
CacheDataset in script not working

### ISSUE THREAD

**Issue by mioDio** [Project-MONAI/MONAI#1616]
Hi, congratulations on a great package, I really like Monai!
I was trying the tutorials and ran the 3D spleen segmentation. One of the features used in the Jupyter notebook was "Cache Dataset". It works fine with my dataset in the notebook, but then I re-wrote it into a script and it is throwing an error, as shown below. The issue is clearly with the Caching, since it works when I use just the "Dataset" module. Further, I noticed that you use the Dataset in your tutorial script as well and I was wondering if there is an underlying reason for that.
Your help would be very much appreciated
S.

`Traceback (most recent call last):a:  [=====                         ]
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/utils.py", line 309, in apply_transform
    return transform(data)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/io/dictionary.py", line 77, in __call__
    data = self._loader(d[key], reader)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/io/array.py", line 103, in __call__
    img = reader.read(filename)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/data/image_reader.py", line 145, in read
    img_.append(itk.imread(name, **kwargs_))
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkExtras.py", line 628, in imread
    TemplateReaderType = itk.ImageFileReader
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkLazy.py", line 52, in __getattribute__
    itkBase.LoadModule(module, namespace)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkBase.py", line 61, in LoadModule
    swig.update(this_module.swig)
AttributeError: module 'itk.ITKIOImageBasePython' has no attribute 'swig'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "Unet_v1_train.py", line 229, in <module>
    main(data_dir, max_epochs=100, num_samples=10, num_val=4)
  File "Unet_v1_train.py", line 118, in main
    train_ds = CacheDataset(
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/data/dataset.py", line 278, in __init__
    p.map(
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 768, in get
    raise self._value
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/data/dataset.py", line 309, in _load_cache_item_thread
    self._cache[i] = self._load_cache_item(item, transforms)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/data/dataset.py", line 297, in _load_cache_item
    item = apply_transform(_transform, item)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/utils.py", line 311, in apply_transform
    raise type(e)(f"Applying transform {transform}.").with_traceback(e.__traceback__)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/utils.py", line 309, in apply_transform
    return transform(data)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/io/dictionary.py", line 77, in __call__
    data = self._loader(d[key], reader)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/transforms/io/array.py", line 103, in __call__
    img = reader.read(filename)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/monai/data/image_reader.py", line 145, in read
    img_.append(itk.imread(name, **kwargs_))
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkExtras.py", line 628, in imread
    TemplateReaderType = itk.ImageFileReader
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkLazy.py", line 52, in __getattribute__
    itkBase.LoadModule(module, namespace)
  File "/home/ubuntu/envs/monai/lib/python3.8/site-packages/itkBase.py", line 61, in LoadModule
    swig.update(this_module.swig)
AttributeError: Applying transform <monai.transforms.io.dictionary.LoadImaged object at 0x7f7703fa9d30>.`

**Comment by rijobro** [comment]
So you're saying that the exact same code runs with `Dataset` but not with `CacheDatset`? Strange! It looks like the error is related to swig, which might be solved with `pip install --upgrade swig`. If not, would it be possible to create a minimum working example? You should be able to create a dictionary containing just 1 image, something like this: 

```python
im_fname, seg_fname = [make_nifti_image(i) for i in create_test_image_2d(101, 100)]
data = [{"im": im_fname, "seg": seg_fname}]
transforms = Compose([
    LoadImaged(...),
    AddChanneld(...),
])
ds = DataSet(...)  # or CacheDataset
dl = DataLoader(...)
next(iter(dl))
```

**Comment by mioDio** [comment]
Hi, 

thank you for the quick answer. I solved the problem by upgrading to MONAI-0.4.0.
The problem occurred with 0.3.0. Not sure if the swig route would have helped, I didn't get to try it, since
I upgraded Monai before that. 

Also, along the way I discovered that Monai 0.3.0 had an issue with Nifti orientation;
"SAR" versus "RAS". I ran two systems with different versions of Monai without knowing it and got perplexed why I didn't get the same transforms. Maybe worth mentioning.

Kind regards and thank you again for your help
